src/View/View.php

```diff
@@ -27,6 +28,116 @@ class View extends BaseView
      */
     protected array $fragmentStack = [];
 
+    /**
+     * Starts holds content for a fragment within the layout.
+     *
+     * @param string $name Fragment name
+     */
+    public function fragment(string $name): void
+    {
+        $this->fragmentStack[] = $name;
+
+        if ($this->showFragments) {
+            echo sprintf('@[[fragmentStart="%s"]]', $name);
+        }
+    }
+
+    /**
+     * Captures the last fragment
+     *
+     * @throws RuntimeException
+     */
+    public function endFragment(): void
+    {
+        if ($this->fragmentStack === []) {
+            ob_end_clean();
+
+            throw new RuntimeException('View themes, no current fragment.');
+        }
+
+        $name = array_pop($this->fragmentStack);
+
+        if ($this->showFragments) {
+            echo sprintf('@[[fragmentEnd="%s"]]', $name);
+        }
+    }
+
+    /**
+     * Whether we should display fragments tags or not.
+     */
+    protected function showFragments(bool $display = true): RendererInterface
```

Comment:
Why not `self` or `static`?

Reply:
I wanted to follow the current convention used in the original `View` class: https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/View/View.php

Reply:
Good. This is a notification in the IDE